### PR TITLE
Fixes follow up to #17723: Notification ads are not shown on Linux

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/17813

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See https://github.com/brave/brave-browser/issues/17813. Needs testing on all platforms.

### CODE REVIEW

**Review by emerick — APPROVED** [review]

**Review by aseren — APPROVED** [review]

**Comment by tmancey** [comment]
CI passed for all platforms except Windows. Fixing Windows specific code and restarting CI.

File: components/brave_ads/browser/notification_helper_android.cc [hunk 1]
```diff
@@ -22,25 +20,45 @@ namespace brave_ads {
...
-  if (features::IsCustomAdNotificationsEnabled()) {
+bool IsBraveAdsNotificationChannelEnabled(const bool is_foreground) {
+  if (GetOperatingSystemMajorVersion() <
+      kMinimumMajorOperatingSystemVersionForNotificationChannels) {
     return true;
   }
 
-  return CanShowNativeNotifications();
+  JNIEnv* env = base::android::AttachCurrentThread();
+  auto j_channel_id = (is_foreground)
```
Comment (moritzhaller) [inline components/brave_ads/browser/notification_helper_android.cc:46]: nit: maybe const?

File: components/brave_ads/browser/notification_helper_android.cc [hunk 2]
```diff
@@ -22,25 +20,45 @@ namespace brave_ads {
...
+                          ? Java_BraveAds_getBraveAdsChannelId(env)
+                          : Java_BraveAds_getBraveAdsBackgroundChannelId(env);
+  auto status = static_cast<NotificationChannelStatus>(
```
Comment (moritzhaller) [inline components/brave_ads/browser/notification_helper_android.cc:49]: nit: maybe const?

**Review by moritzhaller — APPROVED** [review]

File: components/brave_ads/browser/notification_helper_android.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc:46]: Fixed

File: components/brave_ads/browser/notification_helper_android.cc (hunk 2 above)
Comment (tmancey) [reply components/brave_ads/browser/notification_helper_android.cc:49]: Fixed

**Comment by btlechowski** [comment]
LGTM

Verification passed on

Brave | 1.31.23 Chromium: 93.0.4577.63 (Official Build) nightly (64-bit)
-- | --
Revision | ff5c0da2ec0adeaed5550e6c7e98417dac77d98a-refs/branch-heads/4577@{#1135}
OS | Ubuntu 18.04 LTS

Verified system notifications are shown by default
image

Verified custom ads notifications are shown when overridden through variations.

**Pull request merged**